src/virtual-machine.js

```diff
@@ -150,17 +150,27 @@ VirtualMachine.prototype.postIOData = function (device, data) {
 /**
  * Load a project from a Scratch 2.0 JSON representation.
  * @param {?string} json JSON string representing the project.
+ * @return {!Promise} Promise that resolves after targets are installed.
  */
 VirtualMachine.prototype.loadProject = function (json) {
-    this.clear();
+    var that = this;
```

Comment:
Please use `.bind(this)` on the callback function (this will make the merge easier with the ES6 PR).